internal/backends/dotnet/project_files.go

```diff
@@ -0,0 +1,143 @@
+package dotnet
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/replit/upm/internal/api"
+	"github.com/replit/upm/internal/util"
+)
+
+const lockFileName = "packages.lock.json"
+
+// an individual package record from .csproj file
+type packageReference struct {
+	XMLName xml.Name `xml:"PackageReference"`
+	Include string   `xml:"Include,attr"`
+	Version string   `xml:"Version,attr"`
+}
+
+// .csproj file structure
+type project struct {
+	XMLName  xml.Name           `xml:"Project"`
+	Packages []packageReference `xml:"ItemGroup>PackageReference"`
+}
+
+// looks for the .csproj file in the current directory
+func findSpecFile() string {
+	files, err := ioutil.ReadDir("./")
+	if err != nil {
+		util.Die("Can't read current directory")
+	}
+
+	for _, f := range files {
+		if strings.HasSuffix(f.Name(), ".csproj") {
+			return f.Name()
+		}
+	}
+
+	return ".csproj"
+}
+
+// loads the details of the project spec file
+func listSpecfile() map[api.PkgName]api.PkgSpec {
+	var pkgs map[api.PkgName]api.PkgSpec
+	projectFile := findSpecFile()
+	if util.Exists(projectFile) {
+
+		specReader, err := os.Open(projectFile)
+		if err != nil {
```

Comment:
```suggestion
	specReader, err := os.Open(projectFile)
	if errors.Is(err, os.ErrNotExist) {
		return pkgs
	}
	if err != nil {
		util.Die("Could not open %s, with error: %q", projectFile, err)	
	}
	defer specReader.Close()
```

this way we only do one syscall (avoiding potential races, known as [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use)) _and_ we use the [guard clause pattern](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) at the same time.